beacon/sync/src/main/java/tech/pegasys/teku/beacon/sync/forward/multipeer/batches/SyncSourceBatch.java

```diff
@@ -299,6 +348,58 @@ public String toString() {
         .toString();
   }
 
+  private boolean validateNewBlocks(final List<SignedBeaconBlock> newBlocks) {
+    if (blocks.isEmpty() || newBlocks.isEmpty()) {
+      return true;
+    }
+    final SignedBeaconBlock previousBlock = blocks.get(blocks.size() - 1);
+    final SignedBeaconBlock firstNewBlock = newBlocks.get(0);
+    if (!firstNewBlock.getParentRoot().equals(previousBlock.getRoot())) {
+      LOG.debug(
+          "Marking batch invalid because new blocks do not form a chain with previous blocks");
+      return false;
+    }
+    return true;
+  }
+
+  private boolean validateNewBlobSidecars(
+      final List<SignedBeaconBlock> newBlocks,
+      final Map<Bytes32, List<BlobSidecar>> newBlobSidecarsByBlockRoot) {
+    for (final SignedBeaconBlock block : newBlocks) {
```

Comment:
I have doubts here. Seems like you are validating blobs from the blocks we have.

what if:
blockByRange(10,12) returns:
`[blockAt10,BlockAt12]`

BlobsByRange(10,12) returns:
`[BlobsAt10[],BlobsAt11[],BlobsAt12[]]`

and BlobsAt11 contain a sidecar with slot set to `100000`

that sidecar isn't validated and end up being imported and never pruned (slot `100000`)

Maybe we just need to check that `newBlobSidecarsByBlockRoot.size() == newBlocks.size()`
and we need to make sure we have newBlobSidecarsByBlockRoot populated also for blocks with no sidecars. Or, you should track the sidecars you visited and make sure you haven't left any out.

Reply:
Wait, you then import sidecars by block so they wont be considered. Sorry :-)
there is still a  weird situation in which the peer sent you random blobs uncorrelated with blocks and we don't care. We could make an argument that we should downscore it, but it is non critical...

Reply:
Good point, It makes sense. I added a commit with this change and a test. Debug log looks like this:
```
2023-03-23 18:48:15.468+00:00 | ForkJoinPool-1-worker-1 | DEBUG | SyncSourceBatch | Unexpected blob sidecars with roots [0x7e3c83fd2c8dab66478a606a6f1a996132f655886c22be380f7b9eac6631682d] were received
```